cibuildwheel/ios.py

```
@@ -0,0 +1,603 @@
+                elif config.arch != os.uname().machine:
+                    log.step("Skipping tests on non-native simulator architecture")
+                else:
+                    if build_options.before_test:
+                        before_test_prepared = prepare_command(
+                            build_options.before_test,
+                            project=".",
+                            package=build_options.package_dir,
+                        )
+                        shell(before_test_prepared, env=env)
+
+                    log.step("Setting up test harness...")
+                    # Clone the testbed project into the build directory
+                    testbed_path = identifier_tmp_dir / "testbed"
+                    call(
+                        "python",
+                        host_install_path / "testbed",
+                        "clone",
+                        testbed_path,
+                        env=build_env,
+                    )
+
+                    if build_options.test_sources:
+                        copy_test_sources(
+                            build_options.test_sources,
+                            build_options.package_dir,
+                            testbed_path / "iOSTestbed" / "app",
+                        )
+                    else:
+                        # There is no explicit list of test sources. Copy *all*
+                        # the test sources; however use the sdist to do this so
+                        # that we avoid copying any .git or venv folders.
+
+                        # Build a sdist of the project
+                        call(
+                            "python",
+                            "-m",
+                            "build",
+                            build_options.package_dir,
+                            "--sdist",
+                            f"--outdir={identifier_tmp_dir}",
+                            capture_stdout=True,
+                        )
```

This fails with `python` not found in PATH. If the build frontend is build (or build[uv]), it could succeed adding `env=build_env,` but will fail with `pip` frontend.

---

Good catch. I've defined `test-sources` in all my practical test cases; so this branch isn't getting tested.

Adding `env=build_env` is an obvious fix for the `build` case; but I can't see an obvious fix for the `pip` case. A literal "cp -r *" would *work* - but in my experience its a real problem in practice, because that will copy `.git` and `.venv` folders, which (1) take a long time to copy, and (2) can cause problems with building if there are any local binaries. 

I'm wonder if it might be better to just raise an error on this branch of the code. `test-sources` really is the best option for iOS projects, as you copy over *just* the test code into the test app; this branch is really a "fallback to make things easy as a default". But if the default won't actually be that easy.

There's a middle ground where we raise an error for the `pip` build case, but allow an sdist build for the `build` case... but I'm not sure if it's worth carrying that extra code just to make a subset of one case a little easier.

---

If overall we want to always use test-sources (even if only an override), then we're probably better off raising a configuration error in this case.

---

To clarify -  are you suggesting a configuration error for *any* "`tests-sources` is undefined" case, or only for the "`test-sources` is undefined and build backend is pip" case?

---

configuration error for any "tests-sources is undefined" case